GMailSync.txt

```diff
@@ -0,0 +1,34 @@
+Gmail Integration into James
+
+Gmail do that with Gmailify, but it doesn't apply to any IMAP account.
+
+Gmail does synchronization to provide search capabilities, sorting, etc, it's not a backend but rather a bidirectional sync tool.
+
+For James, it makes sense to be able to configure sync processes per user.
+
+It could live outside James process and access James from the outside but it would require talking IMAP or JMAP, non of which is good enough for efficient sync purpose.
```

Comment:
> It could live outside James process and access James from the outside

So what's the difference with IMAPsync?

> non of which is good enough for efficient sync purpose

can you explain why?

Reply:
from imapsync documentation : 

> Imapsync is not adequate for maintaining two active imap accounts in
    synchronization when the user plays independently on both sides. Use
    offlineimap (written by John Goerzen) or mbsync (written by Michael R.
    Elkins) for a 2 ways synchronization.

Then : 

> OfflineIMAP is software that downloads your email mailbox(es) as local Maildirs. OfflineIMAP will synchronize both sides via IMAP.

And : 

> mbsync : Synchronization is based on unique message identifiers (UIDs), so no identification conflicts can occur (as opposed to some other mail synchronizers). Synchronization state is kept in one local text file per mailbox pair; multiple replicas of a mailbox can be maintained. 

This last one could be good enough but I would not bet on it, it's an old C server hosted on sourceforge. At least it would not allow modern monitoring, metrics et al.

Reply:
> none of which is good enough for efficient sync purpose

- we don't have a good IMAP client library (it's an old topic but never sorted out)
- we don't have a JMAP java client library (could be written easily)
- translating changes from GMail to JMAP is harder than accessing mailbox-api because JMAP is building a domain layer on top of mailbox-api that would come in the way for this task

All solutions are doable, there's a more efficient one, IMO.